feature/policy_forwarding/otg_tests/mpls_gre_ipv4_encap_scale_test/mpls_gre_ipv4_encap_scale_test.go

```diff
@@ -0,0 +1,616 @@
+		LagID:       1,
+		IsLag:       true,
+	}
+	agg2 = &otgconfighelpers.Port{
+		Name:        "Port-Channel2",
+		AggMAC:      "02:00:01:01:01:01",
+		MemberPorts: []string{"port3", "port4"},
+		Interfaces:  []*otgconfighelpers.InterfaceProperties{agg2Intf},
+		LagID:       2,
+		IsLag:       true,
+	}
+
+	agg2Intf = &otgconfighelpers.InterfaceProperties{
+		IPv4:        "194.0.2.2",
+		IPv6:        "2001:10:1:6::2",
+		IPv4Gateway: "194.0.2.1",
+		IPv6Gateway: "2001:10:1:6::1",
+		Name:        "Port-Channel2",
+		MAC:         "02:00:01:01:01:02",
+		IPv4Len:     29,
+		IPv6Len:     126,
+	}
+
+	// Custom IMIX settings for all flows.
+	sizeWeightProfile = []otgconfighelpers.SizeWeightPair{
+		{Size: 64, Weight: 20},
+		{Size: 128, Weight: 20},
+		{Size: 256, Weight: 20},
+		{Size: 512, Weight: 10},
+		{Size: 1500, Weight: 30},
+	}
+	flowResolveArp = &otgvalidationhelpers.OTGValidation{
+		Interface: &otgvalidationhelpers.InterfaceParams{Names: []string{agg2.Name}},
+	}
+	// MPLSOGRE Encap IPv4 interface IPv4 Payload
+	flowIPv4 = &otgconfighelpers.Flow{
+		TxNames:           []string{},
+		RxNames:           []string{agg2.Name + ".IPv4"},
+		SizeWeightProfile: &sizeWeightProfile,
+		Flowrate:          40,
+		FlowName:          "GCI traffic IPv4 interface IPv4 Payload",
+		EthFlow:           &otgconfighelpers.EthFlowParams{SrcMAC: agg1.AggMAC},
+		VLANFlow:          &otgconfighelpers.VLANFlowParams{VLANId: 1},
+		IPv4Flow:          &otgconfighelpers.IPv4FlowParams{IPv4Src: "12.1.1.1", IPv4Dst: "194.0.2.2", IPv4SrcCount: 100},
+	}
+
+	flowIPv4Validation = &otgvalidationhelpers.OTGValidation{
+		Interface: &otgvalidationhelpers.InterfaceParams{Names: []string{agg2.Name}, Ports: append(agg1.MemberPorts, agg2.MemberPorts...)},
+		Flow:      &otgvalidationhelpers.FlowParams{Name: flowIPv4.FlowName, TolerancePct: 0.5},
+	}
+
+	// MPLSOGRE Encap IPv6 interface IPv6 Payload
+	flowIPv6 = &otgconfighelpers.Flow{
+		TxNames:   []string{},
+		RxNames:   []string{agg2.Name + ".IPv6"},
+		FrameSize: 1500,
+		Flowrate:  40,
+		FlowName:  "GCI traffic IPv6 interface IPv6 Payload",
+		EthFlow:   &otgconfighelpers.EthFlowParams{SrcMAC: agg1.AggMAC},
+		VLANFlow:  &otgconfighelpers.VLANFlowParams{VLANId: 1},
+		IPv6Flow:  &otgconfighelpers.IPv6FlowParams{IPv6Src: "3000:1::1", IPv6Dst: "2000:1::1", IPv6SrcCount: 100, TrafficClass: 0, TrafficClassCount: 100},
+	}
+
+	flowIPv6Validation = &otgvalidationhelpers.OTGValidation{
+		Interface: &otgvalidationhelpers.InterfaceParams{Names: []string{agg2.Name}, Ports: append(agg1.MemberPorts, agg2.MemberPorts...)},
+		Flow:      &otgvalidationhelpers.FlowParams{Name: flowIPv6.FlowName, TolerancePct: 0.5},
+	}
+	outerGREIPLayerIPv4 = &packetvalidationhelpers.IPv4Layer{
+		DstIP:    "10.99.1.19",
+		Protocol: greProtocol,
+		Tos:      96,
+		TTL:      64,
+	}
+	mplsLayer = &packetvalidationhelpers.MPLSLayer{
+		Label: 19016,
+		Tc:    1,
+	}
+	innerIPLayerIPv4 = &packetvalidationhelpers.IPv4Layer{
```

Comment:
nit: `IP` and `IPv4` in name are redundant.


```suggestion
	innerLayerIPv4 = &packetvalidationhelpers.IPv4Layer{
```